arithmetic_analysis/newton_raphson.py

```diff
@@ -1,37 +1,60 @@
 # Implementing Newton Raphson method in Python
 # Author: Syed Haseeb Shah (github.com/QuantumNovice)
 # The Newton-Raphson method (also known as Newton's method) is a way to
-# quickly find a good approximation for the root of a real-valued function
+# quickly find a good approximation for the root of a real-valued function.
+# https://en.wikipedia.org/wiki/Newton%27s_method
 from __future__ import annotations
 
-from decimal import Decimal
-from math import *  # noqa: F403
-
-from sympy import diff
+from sympy import diff, symbols, sympify
 
 
 def newton_raphson(
-    func: str, a: float | Decimal, precision: float = 10**-10
+    func: str, start_point: float, precision: float = 10**-10
 ) -> float:
     """Finds root from the point 'a' onwards by Newton-Raphson method
     >>> newton_raphson("sin(x)", 2)
     3.1415926536808043
     >>> newton_raphson("x**2 - 5*x +2", 0.4)
-    0.4384471871911695
+    0.4384471871911696
     >>> newton_raphson("x**2 - 5", 0.1)
     2.23606797749979
     >>> newton_raphson("log(x)- 1", 2)
     2.718281828458938
+    >>> from scipy.optimize import newton
+    >>> all(newton_raphson("log(x)- 1", 2) == newton("log(x)- 1", 2)
+    ...     for precision in 10, 100, 1000, 10000))
+    True
+    >>> newton_raphson("log(x)- 1", 2, 0)
+    Traceback (most recent call last):
+        ...
+    ValueError: precision must be greater than zero
+    >>> newton_raphson("log(x)- 1", 2, -1)
+    Traceback (most recent call last):
+        ...
+    ValueError: precision must be greater than zero
     """
-    x = a
-    while True:
-        x = Decimal(x) - (
-            Decimal(eval(func)) / Decimal(eval(str(diff(func))))  # noqa: S307
-        )
-        # This number dictates the accuracy of the answer
-        if abs(eval(func)) < precision:  # noqa: S307
+    x = start_point
```

Comment:
if precision <= 0 then raise a ValueError with the message that matches the test above.